### PR TITLE
Add Response::withCookieCollection().

### PR DESCRIPTION
**Pull request by ADmad**
Refs #12551.

The new method can be added in 3.8 itself instead of 4.0.

### CODE REVIEW

File: src/Http/Response.php [hunk 1]
```diff
@@ -2387,6 +2387,20 @@ public function getCookieCollection()
         return $this->_cookies;
     }
 
+    /**
+     * Get a new instance with provided cookie collection.
+     *
+     * @param \Cake\Http\Cookie\CookieCollection $cookieCollection Cookie collection to set.
+     * @return static
+     */
+    public function withCookieCollection(CookieCollection $cookieCollection)
```
Comment (markstory) [inline src/Http/Response.php]: Should this hint be for the interface?

Comment (ADmad) [reply src/Http/Response.php]: `CookieCollection` doesn't implement any interface besides `IteratorAggregate` and `Countable`.

Comment (markstory) [reply src/Http/Response.php]: My bad. I thought we had interfaces.

**Comment by markstory** [comment]
Docs updated in cakephp/docs@64781fe0a4d4b55a452fbc3144309bff0c87ea01

**Pull request merged**